### PR TITLE
ElasticSearch operator upgrade created new PVCs

### PR DESCRIPTION
**Pull request by mburke5678**
https://bugzilla.redhat.com/show_bug.cgi?id=1868300

### CODE REVIEW

**Comment by mburke5678** [comment]
@QiaolingTang Can you please review this PR? Thank you.

**Comment by mburke5678** [comment]
@vikram-redhat @sichvoge Approved adding this information to the logging docs, as it is important and affected a number of customers.  You suggested the information also go in the release notes.
I thought we should add the information to the 4.5 release notes as a Known Issue, as that is likely where users would look for changes before updating. 
Should we (also/instead) add to the _Asynchronous errata updates_ in 4.4, where the issue was fixed? Users likely wouldn't look there prior to updating to 4.5, and after the user updates, the issue will not present itself when updating to 4.5. 

cc @codyhoag @jeana-redhat

File: modules/cluster-logging-updating-logging.adoc [hunk 1]
```diff
@@ -32,6 +32,21 @@ If your cluster logging version is prior to 4.4, you must upgrade cluster loggin
 
 * Back up your Elasticsearch and Kibana data.
 
+* If your internal Elasticsearch instance uses persistent volume claims (PVC), the PVCs must contain a `logging-cluster:elasticsearch` label. If you are upgrading from an {product-title} version prior to version 4.4.30, you must manually add the label to the Elasticsearch PVCs or the PVCs get removed by the garbage collection process and recreated by the Elasticsearch operator during the upgrade. After {product-title} 4.4.30, the Elasticsearch operator automatically adds the label when it creates the PVC. 
++
+For example:
```
Comment (QiaolingTang) [inline modules/cluster-logging-updating-logging.adoc:37]: @mburke5678 how about using `oc` command here?
```
oc label pvc --all -n openshift-logging logging-cluster=elasticsearch
```
or add label to PVC one by one:
```
oc label pvc $pvc-name -n openshift-logging logging-cluster=elasticsearch
```

Comment (mburke5678) [reply modules/cluster-logging-updating-logging.adoc:37]: @QiaolingTang Am I doing something wrong? I cannot label the PVCs. 
If I use the `oc label pvc` command, the PVC gets labeled, but it is immediately removed.
If I edit the PVC in the console or CLI, I get an error:

`PersistentVolumeClaim in version "v1" cannot be handled as a PersistentVolumeClaim: v1.PersistentVolumeClaim.ObjectMeta: v1.ObjectMeta.Labels: ReadString: expects " or n, but found t, error found in #10 byte of ...|csearch":true}},"spe|..., bigger context ...|logging-cluster":"elasticsearch","elasticsearch":true}},"spec":{"accessModes":["ReadWriteOnce"],"res|...`

Comment (QiaolingTang) [reply modules/cluster-logging-updating-logging.adoc:37]: @mburke5678 which version of EO did you deployed? I tried several times in my cluster and found if the EO has the bug fix, after adding labels to the PVCs, the EO rewrites the PVC label to `logging-cluster=elasticsearch`. If scale down the EO, then the labels can be added successfully

Comment (mburke5678) [reply modules/cluster-logging-updating-logging.adoc:37]: @QiaolingTang I used a 4.6 and 4.4.30 (which has the fix). I didn't realize that the EO rewrites the labels. I will try an earlier version. Thank you.

File: modules/cluster-logging-updating-logging.adoc [hunk 2]
```diff
@@ -32,6 +32,18 @@ If your cluster logging version is prior to 4.4, you must upgrade cluster loggin
 
 * Back up your Elasticsearch and Kibana data.
 
+* If your internal Elasticsearch instance uses persistent volume claims (PVC), the PVCs must contain a `logging-cluster:elasticsearch` label. Without the label, during the upgrade the garbage collection process removes those PVCs and the Elasticsearch Operator creates new PVCs.
```
Comment (bobfuru) [inline modules/cluster-logging-updating-logging.adoc:35]: s/(PVC), the/(PVCs), the/

File: modules/cluster-logging-updating-logging.adoc [hunk 3]
```diff
@@ -32,6 +32,18 @@ If your cluster logging version is prior to 4.4, you must upgrade cluster loggin
...
+** If you are updating from an {product-title} version prior to version 4.4.30, you must manually add the label to the Elasticsearch PVCs. 
++
+For example, you can use the following command to add a label to all the Elasticsearch PVCs:
++
+[source,terminal]
+----
+$ oc label pvc --all -n openshift-logging logging-cluster=elasticsearch
+----
+
+** After {product-title} 4.4.30, the Elasticsearch operator automatically adds the label to the PVCs.
```
Comment (bobfuru) [inline modules/cluster-logging-updating-logging.adoc:45]: s/operator/Operator/

File: release_notes/ocp-4-5-release-notes.adoc [hunk 4]
```diff
@@ -1916,6 +1916,8 @@ In the table below, features are marked with the following statuses:
 [id="ocp-4-5-known-issues"]
 == Known issues
 
+* If your internal Elasticsearch instance uses persistent volume claims (PVC), the PVCs must contain a `logging-cluster:elasticsearch` label. Without the label, during the upgrade the garbage collection process removes those PVCs and the Elasticsearch Operator creates new PVCs. If you are updating from an {product-title} version prior to version 4.4.30, you must manually add the label to the Elasticsearch PVCs. After {product-title} 4.4.30, the Elasticsearch operator automatically adds the label to the PVCs.
```
Comment (bobfuru) [inline release_notes/ocp-4-5-release-notes.adoc:1919]: s/(PVC), the/(PVCs), the/

Comment (bobfuru) [reply release_notes/ocp-4-5-release-notes.adoc:1919]: s/operator/Operator/

**Review by bobfuru — COMMENTED** [review]
A few comments and otherwise LGTM.

**Pull request merged**